### PR TITLE
chore(test_util): replace tempdir code w/tempdir crate

### PR DESCRIPTION
**Pull request by mmastrac**

### CODE REVIEW

**Review by dsherret — CHANGES_REQUESTED** [review]
This was done because TempDir was insanely slow (probably because it is secure) on my old hard drive in Windows (I have not tested on my new one). The tests would take way longer to run with it. See https://github.com/denoland/deno/pull/14153

Edit: I just realized this is tempdir and not tempfile. That said, it looks like tempdir is deprecated by tempfile (https://crates.io/crates/tempdir).

**Comment by mmastrac** [comment]
> This was done because TempDir was insanely slow (probably because it is secure) on my old hard drive in Windows (I have not tested on my new one). The tests would take way longer to run with it. See #14153
> 
> Edit: I just realized this is tempdir and not tempfile. That said, it looks like tempdir is deprecated by tempfile (https://crates.io/crates/tempdir).

Ah, thanks for the feedback. Let me experiment w/tempfile and see if there's any impact on Windows. If there is significant impact, I can try using tempfile for linux and the old code for windows.

**Comment by dsherret** [comment]
> If there is significant impact, I can try using tempfile for linux and the old code for windows.

What's the reason to use tempfile and have platform specific code?

**Comment by mmastrac** [comment]
> > If there is significant impact, I can try using tempfile for linux and the old code for windows.
> 
> What's the reason to use tempfile and have platform specific code?

If the impact of tempfile is too much on Windows, we could have a #[cfg(windows)] for alternate code. Although I think I could probably re-architect this code a bit to just use a single parent tempdir if that's the case. I might need your help to see what the perf of windows is like w/this crate.

File: Cargo.toml [hunk 1]
```diff
@@ -123,6 +123,7 @@ sha2 = { version = "0.10.6", features = ["oid"] }
 smallvec = "1.8"
 socket2 = "0.4.7"
 tar = "=0.4.38"
+tempfile = "3.4.0"
```
Comment (dsherret) [inline Cargo.toml:126]: In cli/Cargo.toml there is a:

```
secure_tempfile = { version = "=3.4.0", package = "tempfile" } # different name to discourage use in tests
```

Can you remove that entry?

**Review by dsherret — COMMENTED** [review]
I just tested this out and there's no difference in time to run the tests on my new machine.

**Review by dsherret — APPROVED** [review]
LGTM. Hopefully this makes things more stable.

**Pull request merged**